lib/internal/http2/core.js

```diff
@@ -2001,9 +2008,20 @@ class Http2Stream extends Duplex {
   }
 }
 
-function processHeaders(headers) {
-  assertIsObject(headers, 'headers');
-  headers = Object.assign(Object.create(null), headers);
+function processHeaders(oldHeaders) {
+  assertIsObject(oldHeaders, 'headers');
+  const headers = Object.create(null);
+
+  if (oldHeaders !== null && oldHeaders !== undefined) {
+    const hop = hasOwnProperty.bind(oldHeaders);
+    // This loop is here for performance reason. Do not change.
```

Comment:
Can you add a comment referring to the benchmark that is the reason for “Do not change” (or instead of it)?

Reply:
In frankness I can't. I've added the comment because I saw these types of loops being removed from other critical places. They shouldn't unless there is a key reason to do so.

Our benchmarks do not cover why the previous case were slower, and I struggled in creating one.

```js
'use strict'

const fastify = require('fastify')({
  http2: true
})

fastify
  .get('/', function (req, reply) {
    reply.send({})
  })

fastify.listen(3000, err => {
  if (err) throw err
  console.log(`server listening on ${fastify.server.address().port}`)
})
```

I used this command:

```
h2load -c 1 -m 100 -D 10 http://localhost:3000
```

I'm getting 49k req/sec vs 45k req/sec between this PR and master.
(Note that each of the changes matters for a bit of the improvement and taken alone are hard to measure).